packages/wells/src/client/api/wells.ts

```diff
@@ -40,10 +40,24 @@ export class Wells extends AssetsAPI {
         id: asset.id,
         name: asset.name,
         wellHeadLocation: Wells.mapToWellHeadLocation(asset),
+        dataSource: Wells.getLabel('DataSource', asset.labels),
+        block: Wells.getLabel('Block', asset.labels),
+        field: Wells.getLabel('Field', asset.labels),
+        operator: Wells.getLabel('Operator', asset.labels),
       };
     });
   };
 
+  static getLabel(type: string, labels: Label[] | undefined): string | void {
+    if (labels) {
+      for (const element of labels) {
+        if (element.externalId.startsWith(type)) {
+          return element.externalId.split(':')[1];
```

Comment:
I thought I had explained this so poorly, but then you understood it so perfectly! You rock, Dylan!! :rocket:

Reply:
No your explanation was great, made perfect sense!

Reply:
Oh, one thing though - I may be misunderstanding the JS here, but it seems you are allowing each well to have only one operator, one field, one of all things (by returning only the first one found)? One of the reasons we went with labels instead of just adding this to metadata was to allow them to have multiple values (in particular, operator is sometimes overriden by a bunch of sub-values of the operator, and dataSource can be multiple sources that contributed to assemblng this well data). 
So you may need to collect all labels that fall into this here and have the attributes in the well being an array of values

Reply:
What should the behaviour be here when filtering? For example, if a well has 2 operators, do both need to be in the filter for the well to be include or just one?

Reply:
So, you would have a filter somewhat like labels=[X1, X2, X3] and would return an "OR" of them. So if the well had 
 labels [X2, X4], it would return true because ther is one in common. 
But now that I think of it, we can in the first version at least suppose only one of each field for simplicity. As long as the query can filter on multiple. Let's leave it for now and ask Javier and the domain people if they feel it would make sense to have both multiple values in the query and multiple values in the well. For the field and block, it doesn't make sense to have more than one in the well. But for some other things that we may want to represent as labels, it could.

Reply:
In the latest commit the functionality is as you have described it here. Multiple values can be in the well and if one of them is in the query it is included